data-prepper-plugins/otel-logs-source/src/main/java/org/opensearch/dataprepper/plugins/source/otellogs/OTelLogsGrpcService.java

```diff
@@ -77,54 +66,50 @@ public OTelLogsGrpcService(int bufferWriteTimeoutInMillis,
 
     @Override
     public void export(ExportLogsServiceRequest request, StreamObserver<ExportLogsServiceResponse> responseObserver) {
-        requestProcessDuration.record(() -> processRequest(request, responseObserver));
-    }
-
-    private void processRequest(
-            final ExportLogsServiceRequest request, final StreamObserver<ExportLogsServiceResponse> responseObserver) {
         requestsReceivedCounter.increment();
         payloadSizeSummary.record(request.getSerializedSize());
 
-        if (Context.current().isCancelled()) {
-            requestTimeoutCounter.increment();
-            responseObserver.onError(Status.CANCELLED.withDescription("Cancelled by client").asRuntimeException());
+        if (ServiceRequestContext.current().isTimedOut()) {
             return;
         }
-        List<OpenTelemetryLog> logs;
+
+        if (Context.current().isCancelled()) {
+            throw new RequestCancelledException("Cancelled by client");
```

Comment:
QUES: Is there reason to favor throwing exception rather than `responseObserver.onError`?

Reply:
Makes sense. I did not realize it is captured by the exception handler.